RulesNextDate.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// Обработчик для API запроса
+func NextDateHandler(w http.ResponseWriter, r *http.Request) {
+	// Получаем параметры из запроса
+	nowStr := r.FormValue("now")
+	dateStr := r.FormValue("date")
+	repeat := r.FormValue("repeat")
+
+	// Преобразуем строку "now" в тип time
+	now, err := time.Parse("20060102", nowStr)
+	if err != nil {
+		http.Error(w, "Некорректная дата 'now'", http.StatusBadRequest)
+		return
+	}
+
+	// Вызываем функцию NextDate для вычисления следующей даты
+	next, err := NextDate(now, dateStr, repeat)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Отправляем результат в ответ
+	if next != "" {
+		w.Write([]byte(next))
+	} else {
+		w.Write([]byte("Задача выполнена"))
+	}
+}
+
+// NextDate вычисляет следующую дату для задачи в соответствии с указанным правилом повторения.
+func NextDate(now time.Time, date string, repeat string) (string, error) {
+	// Парсим исходную дату из формата "20060102"
+	taskDate, err := time.Parse("20060102", date)
+	if err != nil {
+		return "", fmt.Errorf("некорректный формат даты: %v", err)
```

Comment:
При оборачивании нужно использовать `%w` (wrap), а просто при выводе в консоль `%v`
```suggestion
		return "", fmt.Errorf("некорректный формат даты: %w", err)
```